lib/sensors/temperature.cpp

```diff
@@ -0,0 +1,58 @@
+#include "temperature.hpp"
+
+namespace hyped::sensors {
+
+Temperature::Temperature(hyped::core::ILogger &log, io::I2c &i2c, const std::uint8_t channel)
+    : log_(log),
+      i2c_(i2c),
+      channel_(channel)
+{
+}
+
+Temperature::~Temperature()
+{
+}
+
+std::optional<std::int16_t> Temperature::read()
+{
+  const auto status_check_result = i2c_.readByte(kTemperature, kStatus);
+  if (!status_check_result) {
+    log_.log(hyped::core::LogLevel::kFatal, "Temperature could not read status");
+    return std::nullopt;
+  }
+  if (status_check_result.value() == 0) {
```

Comment:
This comparison doesn't make any sense since it does not signify anything from the datasheet. The status reg can check for an invalid read (a temp reading beyond the range of the sensor) and also check if the sensor is still busy coverting the data. For the first case, the value will be 0x06 and 0x01 for the latter case. You should add these two as two separate failure cases.

Reply:
In fact, we would actually want to handle the case where data is not ready specially since its not really a failure but rather needs us to check with the sensor again. I will look into this after this PR is merged, but do be sure to add both the cases as failure for now.